### ISSUE TITLE
Acces denied opening a notebook from Azure ML

### ISSUE THREAD

**Issue by claudiosartori** [Microsoft/AzureNotebooks#79]
Please consider the template below to file your issue. 

### Description
First, using the link http://aka.ms/edx-dat203.1x-aml I login to Azure ML.
I add an Experiment (Automobile price data) and when I attempt to open a notebook (R or Python), I get the error ACCESS DENIED:
"You do not have access to the specified notebook.
If someone shared this link with you please ask them to share the link to their library instead of 
sharing the link to their running notebook server.

Otherwise check if you are still signed in and please login again if you've been signed out."

Note: Azure ML user -> cpsartori@hotmail.com

### Repro Steps
1) From Azure ML add a new Blank experiment.
Use Automobile price data (Raw) to put on the experiment canvas
2) Right-click on the Experiment
3) Select Open a new Notebook; next select either R or Python 3
4) you get the message ACCESS DENIED:
"You do not have access to the specified notebook.
If someone shared this link with you please ask them to share the link to their library instead of 
sharing the link to their running notebook server.

Otherwise check if you are still signed in and please login again if you've been signed out."
![azureml-notebook-accessdenied-16apr17](https://cloud.githubusercontent.com/assets/27638910/25073401/7694e1aa-22bc-11e7-86b3-b72bcf544151.png)

### Recommended Metadata
The Jupyter Kernel (Python2, Pyhton3, R, N/A):
Occurrence Time:
Web Browser:
LibraryID and Notebook Name (if applicable):
Logs (!cat .nb.log Or: New/Terminal, $cat .nb.log) - attach files if applicable

**Comment by 8forty** [comment]
FWIW I started getting the same error, but only in Chrome.  Edge works fine for me.

**Comment by crwilcox** [comment]
I have tried to reproduce this in Edge and Chrome and I am failing to reproduce this. I attached a gif of my attempt to repro. Are you doing something different?
![using ml](https://cloud.githubusercontent.com/assets/638797/25105202/75054c82-2378-11e7-9b6a-b379b4a1b3e1.gif)

**Comment by 8forty** [comment]
I don't have the tools to make an animation, but in my case I'm trying to open a notebook that I've created.  In fact, when I create a new notebook in Chrome, I get the access denied error when it tries to open.  But the notebook does show up in the list of notebooks.  Note that I'm running the Creators update -- maybe a coincidence, but the problem started after I updated.

**Comment by crwilcox** [comment]
Alright. so you are opening from +New Notebook, not from the open dataset? I will look at that path a bit and see if I can't repro.

If you want the tool, it is free. It used to be on codeplex but now they have their own site: http://www.screentogif.com/

**Comment by crwilcox** [comment]
I am still not seeing a failure at all.  I am also on the Windows 10 Creators Update and using Chrome version 57.0.2987.133 (64-bit). Likely not important, but my specific build of windows is 14393.rs1_release_sec.170327-1835.

![new_notebook_ml](https://cloud.githubusercontent.com/assets/638797/25139900/7409f9b8-2413-11e7-8e4e-43d1d7185721.gif)

**Comment by 8forty** [comment]
Thanks for the pointer to the tool.  Here's what happens when I create a notebook, and as I mentioned I also cannot open a notebook in Chrome (also "Version 57.0.2987.133 (64-bit)") Everything works correctly in Edge.

![aml-notebook-error](https://cloud.githubusercontent.com/assets/1214101/25140773/9a5c515a-241e-11e7-92b1-c07fb215f517.gif)

**Comment by ThePhoebius** [comment]
Strangely enough, I have the same Chrome build as 8forty and don't encounter the issue, but trying the same steps in Vivaldi results in the error. Edge works fine as well.
Running Jupyter locally in Vivaldi and accessing the AzureML-Workspace via Python library also runs fine.

**Comment by crwilcox** [comment]
I wonder if authentication isn't properly flowing for you. If you can launch F12 developer tools, go to network and then click the notebook, you should see an authenticate call which should try to set a cookie. I attached an animation that shows how to check this. 

![cookieset](https://cloud.githubusercontent.com/assets/638797/25144769/daaab080-2423-11e7-82d6-15cdd091f076.gif)

**Comment by ThePhoebius** [comment]
Thanks for the nudge in that direction. It seems I could resolve the issue.
The cookies was set. I then tried disabling any extension, that could interfere with that process, with no result. Finally I disabled the "block third party cookies" setting in the browser itself.
That did the trick and even after enabling all extensions and with "block third party cookies" being enabled again, I still am able to access the notebooks directly from the workspace.
I'm still left wondering if the issue is going to pop up again, since it already used to run without problems. But if it does, I hope my workaround will still apply.

Thanks for your support @crwilcox !

**Comment by 8forty** [comment]
I get a little bit different behavior than your example (notebook automatically opens in a new tab), otherwise I guess it looks similar?
![aml-notebook-open-f12](https://cloud.githubusercontent.com/assets/1214101/25147395/9ffdaec2-2434-11e7-862e-cbec1f37f4d1.gif)

**Comment by ThePhoebius** [comment]
I've got it. It is the third party cookies.
@8forty try adding an exception "[*.]azure.com" in chrome://settings/content . I did so in both browsers and they now seem to be working fine. Even after some time now I still can open notebooks.

**Comment by 8forty** [comment]
@ThePhoebius awesome, that fixed both opening and creating.  Thanks!

**Comment by crwilcox** [comment]
@ThePhoebius  and @8forty , thanks for sticking with us to figure this out. I am happy you are unblocked.

**Comment by claudiosartori** [comment]
Hello guys!
Thanks a lot for your feedback and your help, and sorry for my late response.
I managed to make it work but using the browser IE11 (but I needed to refresh the screen).
I would also like to comment, that I did try again with Chrome, and I am still getting the same error.

On other hand, I am happy that some people got the same error with Chrome.

Take care! Have a great evening!
Claudio
